internal/pkg/manifest/workload_test.go

```diff
@@ -635,3 +635,63 @@ func TestDependency_UnmarshalYAML(t *testing.T) {
 		})
 	}
 }
+
+func TestUnmarshalPublish(t *testing.T) {
+	testCases := map[string]struct {
+		inContent     string
+		wantedPublish PublishConfig
+		wantedErr     error
+	}{
+		"Valid publish yaml": {
+			inContent: `topics:
+  - name: tests
+    allowed_workers:
+      - hello
+`,
+			wantedPublish: PublishConfig{
+				Topics: []Topic{
+					{
+						Name:           aws.String("tests"),
+						AllowedWorkers: []string{"hello"},
+					},
+				},
+			},
+		},
+		"Empty workers don't appear in topic": {
+			inContent: `topics:
+  - name: tests
+`,
+			wantedPublish: PublishConfig{
+				Topics: []Topic{
+					{
+						Name: aws.String("tests"),
+					},
+				},
+			},
+		},
+		"Error when unmarshalable": {
+			inContent: `topics:
+   - name: tests
+    allowed_workers:
+      - hello
+  - name: orders
```

Comment:
```suggestion
   - name: tests
     allowed_workers:
       - hello
  - name: orders
```

Reply:
This is to check that it fails when there's a yaml failure, hence the inaccurate yaml format. Should I just not test that?

Reply:
oooh, good call. Ignore this